queries.sql

```diff
@@ -1,82 +1,47 @@
-ROLLBACK;
-
-/* Verify that all records in the "animals" table still exist */
-SELECT * FROM animals;
-
-/* Delete all animals born after Jan 1st, 2022 */
-DELETE FROM animals WHERE date_of_birth > '2022-01-01';
-
-/* Create a savepoint for the transaction */
-SAVEPOINT DELETE_ANIMALS_DOB;
-
-/* Update all animals' weight to be their weight multiplied by -1 */
-UPDATE animals SET weight_kg = -1 * weight_kg;
-
-/* Rollback to the savepoint */
-ROLLBACK TO DELETE_ANIMALS_DOB;
-
-/* Update all animals' weights that are negative to be their weight multiplied by -1 */
-UPDATE animals SET weight = -1 * weight WHERE weight < 0;
-
-/* Commit the transaction */
-COMMIT;
-
-/* Write queries to answer the following questions */
-
-
-/* How many animals are there? */
-SELECT COUNT(*) FROM animals;
+-- What animals belong to Melody Pond?
+SELECT animals.name 
+FROM animals 
+JOIN owners ON animals.owner_id = owners.id 
+WHERE owners.full_name = 'Melody Pond';
 
-/* How many animals have never tried to escape? */
-SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT animals.name 
+FROM animals 
+JOIN species ON animals.species_id = species.id 
+WHERE species.name = 'Pokemon';
 
-/* What is the average weight of animals? */
-SELECT AVG(weight_kg) FROM animals;
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT owners.full_name, COALESCE(animals.name, 'No animals') 
+FROM owners 
+LEFT JOIN animals ON owners.id = animals.owner_id;
 
-/* Who escapes the most, neutered or not neutered animals? */
- SELECT neutered, AVG(escape_attempts) FROM animals
- GROUP BY neutered;
+-- How many animals are there per species?
+SELECT species.name, COUNT(*) 
+FROM animals 
+JOIN species ON animals.species_id = species.id 
+GROUP BY species.name;
 
-/* What is the minimum and maximum weight of each type of animal? */
-SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
-FROM animals
-GROUP BY species;
+-- List all Digimon owned by Jennifer Orwell.
+SELECT animals.name 
+FROM animals 
+JOIN species ON animals.species_id = species.id 
+JOIN owners ON animals.owner_id = owners.id 
+WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
 
-/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
-SELECT species, AVG(escape_attempts) AS avg_escape_attempts 
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT animals.name 
 FROM animals 
-WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' 
-GROUP BY species;
+JOIN owners ON animals.owner_id = owners.id 
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT owners.full_name, COUNT(*) 
+FROM owners 
+JOIN animals ON owners.id = animals.owner_id 
+GROUP BY owners.full_name 
+ORDER BY COUNT(*) DESC 
+LIMIT 1;
```

Comment:
- [ ] Kindly include the transaction queries from the previous milestones, especially the ones that delete animals born after _**01/01/2022**_. For a reviewer to successfully review this milestone, the previous codes should be present since this is a continuity of previous milestones. The codes written in previous milestones also determine if the queries written would return the correct results. Kindly implement this change. 💻

Reply:
Hi @Gambit142 thank you for taking time to review my project. I have added the code from previous milestones